WordPress/Sniff.php

```diff
@@ -572,6 +572,47 @@ protected function has_whitelist_comment( $comment, $stackPtr ) {
 	}
 
 	/**
+	 * Check if a token is used within a unit test.
+	 *
+	 * Unit test methods are identified as such:
+	 * - Method name starts with `test_`.
+	 * - Method is within a class which either extends WP_UnitTestCase or PHPUnit_Framework_TestCase.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @param int $stackPtr The position of the token to be examined.
+	 *
+	 * @return bool True if the token is within a unit test, false otherwise.
+	 */
+	protected function is_token_in_test_method( $stackPtr ) {
+		// Is the token inside of a function definition ?
+		$functionToken = $this->phpcsFile->getCondition( $stackPtr, T_FUNCTION );
+		if ( false === $functionToken ) {
+			return false;
+		}
+
+		// Does the function name start with 'test_' ?
+		$functionName = $this->phpcsFile->getDeclarationName( $functionToken );
+		if ( 0 !== strpos( $functionName, 'test_' ) ) {
+			return false;
+		}
+
+		// Is this a method inside of a class ?
+		$classToken = $this->phpcsFile->getCondition( $functionToken, T_CLASS );
+		if ( false === $classToken ) {
+			return false;
+		}
+
+		// Does the class extend either of whitelisted test classes ?
+		$extendedClassName = $this->phpcsFile->findExtendedClassName( $classToken );
+		if ( in_array( $extendedClassName, array( 'WP_UnitTestCase', 'PHPUnit_Framework_TestCase' ), true ) ) {
```

Comment:
Agreed. Implementation wise however, this is not as easy as it would seem.

The thing is, as the method is within the `Sniff.php` class, IMHO the class whitelist property should live there too.

However, AFAIK, the property setting from the ruleset needs to be done for each individual sniff. So in this case that would mean that the custom whitelist property would need to be set in the ruleset for any sniff using the `is_token_in_test_method()` method.
Not much of an issue at this moment what with only one sniff using it, but by the time there are three of four sniffs using it, it will start to feel weird having to add the same property for each of them in the ruleset.